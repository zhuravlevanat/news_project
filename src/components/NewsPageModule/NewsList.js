import List from "@material-ui/core/List";
import React from "react";
import Spinner from "../helpers/Spinner";
import NewsPost from "../NewsPost/NewsPost";


const NewsList = ({ news, loading }) => (
    <>
        {loading
            ? <Spinner/>
            : <List>{news.map((item, index) => <NewsPost key={index} item={item}/>)}</List>}
    </>
);

export default NewsList;
