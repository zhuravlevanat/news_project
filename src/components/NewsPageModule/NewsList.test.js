import React from "react";
import { shallow } from 'enzyme';
import NewsList from "./NewsList";


describe('NewsList', () => {
    const props = {
        news: [],
        loading: false,
    };
    describe('News list initial', () => {
        const newsList = shallow(<NewsList {...props} />);
        it('renders properly', () => {
            expect(newsList).toMatchSnapshot();
        });

        it('render initial', () => {
            expect(newsList.find('Spinner')).toHaveLength(0);
            expect(newsList.find(`WithStyles(ForwardRef(List))`)).toHaveLength(1);
        });
    });
    describe('NewsList with loading', () => {
        const nextProps = {
            ...props,
            loading: true,
        };
        const newsList = shallow(<NewsList {...nextProps} />);

        it('renders properly', () => {
            expect(newsList).toMatchSnapshot();
        });

        it('render loading', () => {
            expect(newsList.find('Spinner')).toHaveLength(1);
            expect(newsList.find(`WithStyles(ForwardRef(List))`)).toHaveLength(0);
        });
    });
    describe('NewsList with news', () => {
        const nextProps = {
            ...props,
            news: [{ title: 'hello' }],
            loading: false,
        };
        const list = shallow(<NewsList {...nextProps} />);
        console.log(list.debug());
        it('render loading', () => {
            expect(list.find('NewsPost')).toHaveLength(nextProps.news.length);
        });
    });
});
