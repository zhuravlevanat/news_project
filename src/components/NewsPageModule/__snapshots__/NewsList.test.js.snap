// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsList News list initial renders properly 1`] = `ShallowWrapper {}`;

exports[`NewsList NewsList with loading renders properly 1`] = `ShallowWrapper {}`;
